receive.c

```diff
@@ -109,10 +112,37 @@ static void on_read(int fd, char *buf, ssize_t len, char **buf1, ssize_t *len1)
 				for(i=0;i<n;i++) {
 					struct JsonNode *json = json_decode(array[i]);
 					if(json != NULL) {
-						char *out = json_stringify(json, "\t");
-						printf("%s\n", out);
+						if(filtertype != 0) {
+							int filtered = 0, j = 0;
+							char *protocol = NULL;
+							if(filtertype == EXCLUDE) {
+								filtered = 0;
```

Comment:
Because the condition for the include filtering doesn't work otherwise - it still prints everything. It would be clearer if it was 

```
if(filtertype == INCLUDE) {
        filtered = 1;
}
```